web3/eth.py

```diff
@@ -308,37 +307,18 @@ def uninstallFilter(self, filter_id):
         )
 
     def contract(self,
-                 *args,
+                 address=None,
                  **kwargs):
         ContractFactoryClass = kwargs.pop('ContractFactoryClass', self.defaultContractFactory)
-        contract_name = kwargs.pop('contract_name', None)
-
-        has_address = any((
-            'address' in kwargs,
-            len(args) >= 1 and is_address(args[0]),
-            len(args) >= 2 and is_address(args[1]),
-        ))
-
-        for potential_address in args:
-            validate_address_checksum(potential_address)
-
-        if has_address:
-            if 'address' in kwargs:
-                address = kwargs.pop('address')
-            elif is_address(args[0]):
-                address = args[0]
-            elif is_address(args[1]):
-                address = args[1]
-                kwargs['abi'] = args[0]
+
+        NewContract = ContractFactoryClass.factory(self.web3, **kwargs)
```

Comment:
I'd like to figure out clear terminology for the *class* and *instance*.  I've used the term *contract* to mean *instance* and *factory* to mean *class*.  This name conflicts a bit with that terminology so I'd either like to come up with terminology that we both like or use the existing terminology and keep things consistently named.

Reply:
:+1: for switching to `ContractFactory`